src/modules/content-seeding/news-aggregator.js

```diff
@@ -0,0 +1,290 @@
+
+
+const performReadQuery = promisify(pg.read.bind(pg, config.connStr));
+const performWriteQuery = promisify(pg.write.bind(pg, config.connStr));
+const parseFeed = promisify(feedParser.bind(feedParser));
+
+type Article = {
+	title: string;
+	link: string;
+	content: string;
+	published: Date;
+}
+
+type FeedsObject = {
+	url: string;
+	mtbu: number;
+	lastrequesttime: number;
+	lastupdatetime: number
+};
+
+type RoomSpecificNews = {
+	roomid: string;
+	roomname: string;
+	article: Article;
+	url: string
+};
+
+function articleAge (articlePubDate: number): number {
+	return Date.now() - articlePubDate;
+}
+
+function buildThreads (latestNewsForRooms: Array<RoomSpecificNews>): Thread {
+	const changes = {
+		entities: {}
+	};
+	for (const newsArticle of latestNewsForRooms) {
+		const id = uuid.v4();
+		changes.entities[id] = {
+			id,
+			type: TYPE_THREAD,
+			name: newsArticle.article.title,
+			body: newsArticle.article.link,
+			tags: [ TAG_RSS_NEWS_SEED ],
+			parents: [ newsArticle.roomid ],
+			identities: [],
+			creator: 'belongbot',
+			createTime: Date.now(),
+		};
+	}
+	return changes;
+}
+
+function calculateMTBU (newArticles: Array<Article>): number {
+	let sum = 0;
+	const articleDatesTimeStamps = _.map(newArticles, (article) => {
+		return (new Date(article.published)).getTime();
+	});
+	const sortedArticleTimeStamps = articleDatesTimeStamps.sort().reverse();
+	const recentTimeStamps = sortedArticleTimeStamps.length >= 3 ? sortedArticleTimeStamps.slice(0, 3) : sortedArticleTimeStamps;
+	recentTimeStamps.forEach((timestamp, index) => {
+		if (index === 0) {
+			sum += (Date.now() - timestamp);
+		} else {
+			sum += (recentTimeStamps[index - 1] - timestamp);
+		}
+	});
+	return sum / (recentTimeStamps.length);
+}
+
+function filterFeeds (): Promise<Array<FeedsObject>> {
+	return performReadQuery({
+		$: 'SELECT * FROM feeds WHERE lastrequesttime < (extract(epoch from NOW()) * 1000 - mtbu)'
+	});
+}
+
+function getRoomSpecificNews (): Promise<Array<RoomSpecificNews>> {
+	return performReadQuery({
+		$: `SELECT rooms.id as roomid, rooms.name as roomname, articles.rawjson as article, articles.url as url 
+			From rooms, articles
+			WHERE articles.terms @@ plainto_tsquery(rooms.name)`
+	});
+}
+
+function getMostRecentNewsForRoom (roomSpecificNewsArticles: Array<RoomSpecificNews>): Array<RoomSpecificNews> {
+	const sortedNewsArticles = _.orderBy(roomSpecificNewsArticles, (thread) => {
+		return thread.article.published;
+	}, [ 'desc' ]);
+	return _.uniqBy(sortedNewsArticles, 'roomid');
+}
+
+function hasNewNewsArticle (articlePubDate: number, lastupdatetime: number): boolean {
+	return articlePubDate > lastupdatetime;
+}
+
+function insertIntoFeeds (url: string): Promise<Array<{ rowCount: number }>> {
+	return performWriteQuery([ {
+		$: `INSERT INTO feeds (url) 
+			VALUES ( &{url} )`,
+		url
+	} ]);
+}
+
+function insertNewArticles (url: string, article: Article, contentForLexemes: string): Promise<Array<{ rowCount: number }>> {
```

Comment:
Accept an array of articles rather than parts of a single one, and make a single INSERT with multiple rows.